drs_filer/ga4gh/drs/endpoints/register_access_methods.py

```diff
@@ -0,0 +1,133 @@
+    db_collection = (
+        current_app.config['FOCA'].db.dbs['drsStore'].
+        collections['objects'].client
+    )
+    obj = db_collection.find_one({"id": object_id})
+    if not obj:
+        logger.error(f"DRS object with id: {object_id} not found.")
+        raise ObjectNotFound
+
+    # Set flags and parameters for POST/PUT routes
+    replace = True
+    if access_id is None:
+        replace = False
+        id_length = (
+            current_app.config['FOCA'].endpoints['objects']['id_length']
+        )
+        id_charset: str = (
+            current_app.config['FOCA'].endpoints['objects']['id_charset']
+        )
+        # evaluate character set expression or interpret literal string as set
+        try:
+            id_charset = eval(id_charset)
+        except Exception:
+            id_charset = ''.join(sorted(set(id_charset)))
+
+    # Try to generate unique ID and insert object into database
+    for i in range(retries + 1):
+        logger.debug(f"Trying to insert/update access method: try {i}")
+        # Set or generate object identifier
+        if access_id is not None:
+            data['access_id'] = access_id
+        else:
+            data['access_id'] = generate_id(
+                charset=id_charset,  # type: ignore
+                length=id_length,  # type: ignore
+            )
+        # Replace access method, then return (PUT)
+        if replace:
+            result_replace = db_collection.update_one(
+                filter={'id': object_id},
+                update={
+                    '$set': {
+                        'access_methods.$[element]': data
+                    }
+                },
+                array_filters=[{'element.access_id': data['access_id']}],
+            )
+
+            if(result_replace.modified_count):
+                logger.info(
+                    f"Replaced access method with access_id: "
+                    f"{data['access_id']} of DRS object with id: {object_id}"
+                )
+                break
+
+        # Try inserting the access method incase of POST or incase
+        # no element matches with the filter incase of PUT.
+        result_insert = db_collection.update_one(
+            filter={
+                'id': object_id,
+                'access_methods.access_id': {'$ne': data['access_id']}
+            },
+            update={
+                '$push': {
+                    'access_methods': data
+                }
+            }
+        )
+        if(result_insert.modified_count):
+            logger.info(
+                f"Added access method with access_id: {data['access_id']}"
+                f" to DRS object with id: {object_id}"
+            )
+            break
+    # Access method neither added nor updated.
+    else:
+        logger.error(
+            f"Could not generate unique identifier. Tried {retries + 1} times."
+        )
+        raise InternalServerError
+    return data['access_id']
+
+
+def generate_id(
```

Comment:
I think you can import that function from FOCA now, it should be available in `0.6.0`